### ISSUE TITLE
Standardize escaping of ! parameters: "Never" option?

### ISSUE THREAD

**Issue by jelliottp** [SAP/abap-cleaner#31]
I noticed there is no `Never` option available for this rule.
![image](https://github.com/SAP/abap-cleaner/assets/63249342/18ed608c-023a-4639-a2cf-9b4d292d2210)

Is it truly required to escape parameters that match ABAP keywords? I have been removing these for some time without issue. Perhaps we could add a `Never` option?

**Comment by jmgrassau** [comment]
Hi Josh, 

you're right, you can hardly find an example where this goes wrong. However, try this case (which is complete nonsense, of course) :

```ABAP
    METHODS any_method
      IMPORTING
        exporting TYPE i OPTIONAL
      EXPORTING
        changing  TYPE string
      CHANGING
        raising   TYPE i.
```

Kind regards,
Jörg-Michael

**Comment by jelliottp** [comment]
Thanks for the example. So it seems importing/exporting/changing/returning/raising are the only cases where it's truly a syntax error. 

Perhaps this could be one of the configuration options? Or become the behavior for `only if parameter name is ABAP word` setting?

**Comment by jmgrassau** [comment]
Hi Josh, 

I think adding "never" as a fourth option would be a good solution – or maybe it could be called "only to avoid syntax errors" or "almost never" :-)  

Please let me know if you find more examples of parameter names that trigger syntax errors. Apart from "exporting", "changing" and "raising", ! is apparently also needed for "default", "optional" and "preferred" (and "default" would be a parameter name that I could even imagine to be used in real life). By contrast, "importing", "value" and "reference" work fine.

Kind regards,
Jörg-Michael

**Comment by jelliottp** [comment]
> Hi Josh,
> 
> I think adding "never" as a fourth option would be a good solution – or maybe it could be called "only to avoid syntax errors" or "almost never" :-)
> 
> Please let me know if you find more examples of parameter names that trigger syntax errors. Apart from "exporting", "changing" and "raising", ! is apparently also needed for "default", "optional" and "preferred" (and "default" would be a parameter name that I could even imagine to be used in real life). By contrast, "importing", "value" and "reference" work fine.
> 
> Kind regards, Jörg-Michael

I would support either solution. "Never" _should_ be sufficient; engineers would be already alerted to an issue with the parameter name due to the syntax error, so cleanup really shouldn't be necessary.

**Comment by jmgrassau** [comment]
Hi Josh, 

okay, in the next release, this option will look as follows:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/af5da349-0dfa-469e-b527-05ebe7022345)

The example now shows all critical keywords that I am aware of:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/a684326a-dcab-4747-868b-4790c582ab9d)

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Josh, 

this new option is now released with version 1.3.0!

Kind regards,
Jörg-Michael
